NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -38,23 +52,89 @@ category_show(io::IO, k, v) = print(io, v)
 
 category_show(io::IO, k::Symbol, v) = print(io, "($k=$v,)")
 
+function Base.isless(s1::C, s2::C) where {C<:CategoryProduct}
+  return isless(
+    category_label.(values(categories(s1))), category_label.(values(categories(s2)))
+  )
+end
```

Comment:
What about if the inputs don't have all of the same symmetries, i.e. if it is a `NamedTuple` and some are missing in one of the inputs?

Reply:
Good question. Currently 2 categories need to have the same type.
Another possibility is to allow for missing sectors and replace them by trivial in sorting. The one thing I do not like is the resulting order is no more antisymmetric: one ends up with
```julia
a = sector(A=U1(0))
b = sector(B=SU2(0))
isless(a, b)  # false
isless(b, a)  # false
a == b  # false
```
That is, unless we overload `==` to add missing singlets, however when we discussed this the conclusion was to avoid overloading `==` too much. Even if we extend `==` to missing sectors in `CategoryProduct` only, we still get a non-symmetric `==`, with `U1(0) == sector()`, `sector() == SU2(0)`, but `U1(0) != SU2(0)`

So there seems to be no ideal solution. I think the current situation is the least confusing one.

Reply:
I just realized with the current implementation of `==` , `a==b` is `true`, however I previously removed the possibility to create a sector with an empty `NamedTuple` to keep only one empty sector. The result is
```julia
sector() == a  # false
a == b  # true
```
then for `NamedTuple` I could indeed insert singlets and compare different `NamedTuple`. For the `Tuple` implementation however this is not possible: either `==` is no more symmetric or `isless` is no more antisymmetric, due to `sector`.
  
Looks like either we accept a different behavior between `Tuple` and `NamedTuple` implementations for missing categories, or we cannot compare when the type differs.

Reply:
Implemented as suggested: different behavior between ordered and `NamedTuple` operations for comparisons.